src/multisigSigning.ts

```diff
@@ -0,0 +1,359 @@
+  'Cannot sign a multisig transaction using `signTxn`. Use `partialSignTxn` instead.';
+export const MULTISIG_SIGNATURE_LENGTH_ERROR_MSG =
+  'Cannot add multisig signature. Signature is not of the correct length.';
+const MULTISIG_KEY_NOT_EXIST_ERROR_MSG = 'Key does not exist';
+
+/**
+ * createMultisigTransaction creates a raw, unsigned multisig transaction blob.
+ * @param txn - the actual transaction.
+ * @param version - multisig version
+ * @param threshold - multisig threshold
+ * @param pks - ordered list of public keys in this multisig
+ * @returns encoded multisig blob
+ */
+export function createMultisigTransaction(
+  txn: Transaction,
+  { version, threshold, addrs }: MultisigMetadata
+) {
+  // construct the appendable multisigned transaction format
+  const pks = pksFromAddresses(addrs);
+  const subsigs = pks.map((pk) => ({ pk }));
+
+  const msig: EncodedMultisig = {
+    v: version,
+    thr: threshold,
+    subsig: subsigs,
+  };
+
+  // if the address of this multisig is different from the transaction sender,
+  // we need to add the auth-addr field
+  const msigAddr = addressFromMultisigPreImg({
+    version,
+    threshold,
+    pks,
+  });
+  let sgnr: Address | undefined;
+  if (!txn.sender.equals(msigAddr)) {
+    sgnr = msigAddr;
+  }
+
+  const signedTxn = new SignedTransaction({
+    txn,
+    msig,
+    sgnr,
+  });
+
+  return encoding.encodeMsgpack(signedTxn);
+}
+
+interface MultisigOptions {
+  rawSig: Uint8Array;
+  myPk: Uint8Array;
+}
+
+interface MultisigMetadataWithPks extends Omit<MultisigMetadata, 'addrs'> {
+  pks: Uint8Array[];
+}
+
+/**
+ * createMultisigTransactionWithSignature creates a multisig transaction blob with an included signature.
+ * @param txn - the actual transaction to sign.
+ * @param rawSig - a Uint8Array raw signature of that transaction
+ * @param myPk - a public key that corresponds with rawSig
+ * @param version - multisig version
+ * @param threshold - multisig threshold
+ * @param pks - ordered list of public keys in this multisig
+ * @returns encoded multisig blob
+ */
+function createMultisigTransactionWithSignature(
+  txn: Transaction,
+  { rawSig, myPk }: MultisigOptions,
+  { version, threshold, pks }: MultisigMetadataWithPks
+): Uint8Array {
+  // Create an empty encoded multisig transaction
+  const encodedMsig = createMultisigTransaction(txn, {
+    version,
+    threshold,
+    addrs: pks.map((pk) => new Address(pk)),
+  });
+  // note: this is not signed yet, but will be shortly
+  const signedTxn = encoding.decodeMsgpack(encodedMsig, SignedTransaction);
+
+  let keyExist = false;
+  // append the multisig signature to the corresponding public key in the multisig blob
+  signedTxn.msig!.subsig.forEach((subsig, i) => {
+    if (nacl.bytesEqual(subsig.pk, myPk)) {
+      keyExist = true;
+      signedTxn.msig!.subsig[i].s = rawSig;
+    }
+  });
+  if (!keyExist) {
+    throw new Error(MULTISIG_KEY_NOT_EXIST_ERROR_MSG);
+  }
+
```

Comment:
this misses msig check compare to the [original code](https://github.com/algorand/js-algorand-sdk/pull/862/files#diff-f860886128b9a9712adffc0a93540ef15ec333cedf3a2bc40953698001c74c86L233)
```
  const msigAddr = addressFromMultisigPreImg({
    version,
    threshold,
    pks,
  });
  const senderAddr = signedTxn.txn.snd
    ? new Address(signedTxn.txn.snd)
    : Address.zeroAddress();
  if (!senderAddr.equals(msigAddr)) {
    signedTxn.sgnr = msigAddr.publicKey;
  }
```

Reply:
Yes, I removed this from `createMultisigTransactionWithSignature` because it's already checked in `createMultisigTransaction`, which is called at the beginning of the function.

`createMultisigTransaction`:
https://github.com/algorand/js-algorand-sdk/blob/b864c24ac3007a2bba08f0bad1c98d1617891ebe/src/multisigSigning.ts#L54-L64